File: MiniDOMBase/DMD/Anagrafica/Uffici/CUtentiXUfficioCollection.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica


 

    <Serializable> _
    Public Class CUtentiXUfficioCollection
        Inherits CCollection(Of CUser)

        <NonSerialized> _
        Private m_Ufficio As CUfficio

        Public Sub New()
            Me.m_Ufficio = Nothing
        End Sub

        Public Sub New(ByVal ufficio As CUfficio)
            Me.New()
            Me.Load(ufficio)
        End Sub

        Public ReadOnly Property Ufficio As CUfficio
            Get
                Return Me.m_Ufficio
            End Get
        End Property

        Protected Friend Sub Load(ByVal value As CUfficio)
            If (value Is Nothing) Then Throw New ArgumentNullException("Ufficio")
            MyBase.Clear()
            Me.m_Ufficio = value
            If (GetID(value) = 0) Then Return

            For i As Integer = 0 To Anagrafica.Uffici.UfficiConsentiti.Count - 1
                Dim item As CUtenteXUfficio = Anagrafica.Uffici.UfficiConsentiti(i)
                If (item.IDUfficio = GetID(value)) AndAlso (item.Utente IsNot Nothing) AndAlso (item.Utente.Stato = ObjectStatus.OBJECT_VALID) Then
                    MyBase.Add(item.Utente)
                End If
            Next
        End Sub


    End Class

End Class

File: MiniDOMBase/DMD/Anagrafica/Uffici/UtentiXUfficio/CUtentiXUfficioCollection.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Collezione di utenti per un ufficio
        /// </summary>
        [Serializable]
        public class CUtentiXUfficioCollection 
            : CCollection<Sistema.CUser>
        {
            [NonSerialized] private CUfficio m_Ufficio;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CUtentiXUfficioCollection()
            {
                m_Ufficio = null;
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="ufficio"></param>
            public CUtentiXUfficioCollection(CUfficio ufficio) : this()
            {
                Load(ufficio);
            }

            /// <summary>
            /// Restituisce un riferimento all'ufficio
            /// </summary>
            public CUfficio Ufficio
            {
                get
                {
                    return m_Ufficio;
                }
            }

            /// <summary>
            /// Carica gli utenti
            /// </summary>
            /// <param name="value"></param>
            protected internal void Load(CUfficio value)
            {
                if (value is null)
                    throw new ArgumentNullException("Ufficio");
                
                this.Clear();

                m_Ufficio = value;
                if (DBUtils.GetID(value, 0) == 0)
                    return;
                
                foreach(var item in minidom.Anagrafica.Uffici.UfficiConsentiti)
                {
                    if (   item.IDUfficio == DBUtils.GetID(value, 0) 
                        && item.Utente is object 
                        && item.Utente.Stato == ObjectStatus.OBJECT_VALID
                        )
                    {
                        Add(item.Utente);
                    }
                }
            }
        }
    }
}